src/index.ts

```diff
@@ -22,6 +22,18 @@ import BN from 'bn.js'
 import { osmosis, thorchain } from './amino'
 import * as codecs from './proto'
 
+type AgnosticStdTx = Omit<amino.StdTx, 'msg'> &
```

Comment:
would probably be easier just to have a union of `amino.StdTx | StdTxWithMsgs` or something like that.

`export type StdTxWithMsgs = Omit<amino.StdTx, 'msg'> & { readonly msgs: readonly amino.AminoMsg[] }`